packages/chat-component-bindings/src/messageThreadSelector.ts

```diff
@@ -73,19 +73,19 @@ const extractAttachedFilesMetadata = (metadata: Record<string, string>): FileMet
   }
 };
 
-/* @conditional-compile-remove(teams-inline-images) */
-const extractInlineImageFilesMetadata = (attachments: ChatAttachment[]): FileMetadata[] => {
+/* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+const extractTeamsAttachmentsMetadata = (attachments: ChatAttachment[]): FileMetadata[] => {
   return attachments.map((attachment) => ({
     attachmentType: mapAttachmentType(attachment.attachmentType),
     id: attachment.id,
     name: attachment.name ?? '',
     extension: attachment.contentType ?? '',
-    url: attachment.url,
+    url: attachment.attachmentType === 'file' && attachment.previewUrl ? attachment.previewUrl : attachment.url,
```

Comment:
nitpick: lets have this logic written into a new helper method like `mapAttachmentUrl` or something along those lines. Just so in the future if / when this grows it has a dedicated place for the logic.